_posts/2023-11-23-stork-latest-news.adoc

```diff
@@ -0,0 +1,188 @@
+business logic for decision-making and data expiration.
+
+This enhancement was driven by the specific requirements of Kubernetes service discovery as contacting the cluster too frequently can result in performance
+problems. So, out of the box, Stork Kubernetes service discovery now comes with a tailored cache expiration strategy to keep service instances until an event occurs.
+
+If you would like to do so for your custom service discovery implementations, you need:
+- Extend the CachingServiceDiscovery as mentioned above.
+- Implement the `cache` method where the expiration strategy is defined.
+- Invalidate the cache when the expiration condition evaluates to `true`.
+
+Look at the example bellow:
+
+[source,java]
+----
+package examples;
+
+import io.smallrye.mutiny.Uni;
+import io.smallrye.stork.api.ServiceInstance;
+import io.smallrye.stork.impl.CachingServiceDiscovery;
+import io.smallrye.stork.impl.DefaultServiceInstance;
+import io.smallrye.stork.utils.ServiceInstanceIds;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class CustomExpirationCachedAcmeServiceDiscovery extends CachingServiceDiscovery {
+
+    private final String host;
+    private final int port;
+
+    private AtomicBoolean invalidated = new AtomicBoolean();
+
+    public CustomExpirationCachedAcmeServiceDiscovery(CachedAcmeConfiguration configuration) {
+        super(configuration.getRefreshPeriod()); // (1)
+        this.host = configuration.getHost();
+        this.port = Integer.parseInt(configuration.getPort());
+    }
+
+    @Override
+    public Uni<List<ServiceInstance>> fetchNewServiceInstances(List<ServiceInstance> previousInstances) {
+        // Retrieve services...
+        DefaultServiceInstance instance =
+                new DefaultServiceInstance(ServiceInstanceIds.next(), host, port, false);
+        return Uni.createFrom().item(() -> Collections.singletonList(instance));
+    }
+
+    @Override
+    public Uni<List<ServiceInstance>> cache(Uni<List<ServiceInstance>> uni) {
+        return uni.memoize().until(() -> invalidated.get());
+    }
+
+    //command-based cache invalidation: user triggers the action to invalidate the cache.
+    public void invalidate() {
+        invalidated.set(true);
+    }
+
+}
+----
+
+You can check the https://github.com/smallrye/smallrye-stork/blob/main/service-discovery/kubernetes/src/main/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscovery.java[Kubernetes Service Discovery code] for further details about an event-based invalidation example.
+
+
+== Observability
+
+Observability refers to the ability to understand and gain insights into the internal workings and behaviors of a system through the analysis
+of its external outputs or observations. Stork becomes observable from the 3.6.0 release. This addition brings automated observability to the forefront
```

Comment:
```suggestion
of its external outputs or observations. Stork becomes observable from Quarkus 3.6.0 release. This addition brings automated observability to the forefront
```

I'm assuming this refer to Quarkus versionn rather than Stork as other version numbers mentioned was 1.3 and 1.4